wolfcrypt/src/ed448.c

```diff
@@ -51,6 +51,119 @@
 static const byte ed448Ctx[ED448CTX_SIZE+1] = "SigEd448";
 #endif
 
+
+static int ed448_hash_init(ed448_key* key, wc_Shake *sha)
+{
+    int ret;
+
+    ret = wc_InitShake256(sha, key->heap,
+#if defined(WOLF_CRYPTO_CB)
+                           key->devId
+#else
+                           INVALID_DEVID
+#endif
+        );
+
+#ifdef WOLFSSL_ED448_PERSISTENT_SHA
+    if (ret == 0)
+        key->sha_clean_flag = 1;
+#endif
+
+    return ret;
+}
+
+#ifdef WOLFSSL_ED448_PERSISTENT_SHA
+static int ed448_hash_reset(ed448_key* key)
+{
+    int ret;
+
+    if (key->sha_clean_flag)
+        ret = 0;
+    else {
+        wc_Shake256_Free(&key->sha);
+        ret = wc_InitShake256(&key->sha, key->heap,
+#if defined(WOLF_CRYPTO_CB)
+                              key->devId
+#else
+                              INVALID_DEVID
+#endif
+            );
+        if (ret == 0)
+            key->sha_clean_flag = 1;
+    }
+    return ret;
+}
+#endif /* WOLFSSL_ED448_PERSISTENT_SHA */
+
+static int ed448_hash_update(ed448_key* key, wc_Shake *sha, const byte* data, word32 len)
+{
+#ifdef WOLFSSL_ED448_PERSISTENT_SHA
+    if (key->sha_clean_flag)
+        key->sha_clean_flag = 0;
+#else
+    (void)key;
+#endif
+    return wc_Shake256_Update(sha, data, len);
+}
+
+static int ed448_hash_final(ed448_key* key, wc_Shake *sha, byte* hash, word32 hashLen)
+{
+    int ret = wc_Shake256_Final(sha, hash, hashLen);
+#ifdef WOLFSSL_ED448_PERSISTENT_SHA
+    if (ret == 0)
+        key->sha_clean_flag = 1;
+#else
+    (void)key;
+#endif
+    return ret;
+}
+
+static void ed448_hash_free(ed448_key* key, wc_Shake *sha)
```

Comment:
Function is only used in the case `#ifndef WOLFSSL_ED448_PERSISTENT_SHA`. Recommend not including otherwise.

Reply:
you didn't squint enough.  3 calls `#ifndef`, 1 call `#ifdef` :grin: